### ISSUE TITLE
[BUG] Sigterm does not terminating efb properly, has to use sigkill to exit. 

### ISSUE THREAD

**Issue by terrytw** [ehForwarderBot/ehForwarderBot#247]
Hello respected devs,

I have encountered a problem while running efb framework on a openwrt x86_64 machine. (openwrt version 19.07.7, with python 3.7)
I installed the framework, master channel, slave channel using pip3. 

After initiating using `ehForwarderBot` command, the QR code showed up, I tried to use ctrl+c to terminate the process without success, pressing ctrl+c for 5 times, the process was killed eventually. 

I followed the instructions, installed thread tracing utility, and here is the report that's coming out of it:
```
---------Thread ************* "MainThread" hangs ---------
        File "/usr/bin/ehforwarderbot", line 8, in <module>
                sys.exit(main())
        File "/usr/lib/python3.7/site-packages/ehforwarderbot/__main__.py", line 339, in main
                init(conf)
        File "/usr/lib/python3.7/site-packages/ehforwarderbot/__main__.py", line 135, in init
                coordinator.add_channel(cls(instance_id=instance_id))
        File "/usr/lib/python3.7/site-packages/efb_wechat_slave/__init__.py", line 157, in __init__
                self.authenticate('console_qr_code', first_start=True)
        File "/usr/lib/python3.7/site-packages/efb_wechat_slave/__init__.py", line 666, in authenticate
                start_immediately=not first_start)
        File "/usr/lib/python3.7/site-packages/efb_wechat_slave/vendor/wxpy/api/bot.py", line 88, in __init__
                loginCallback=login_callback, exitCallback=logout_callback
        File "/usr/lib/python3.7/site-packages/efb_wechat_slave/vendor/itchat/components/register.py", line 33, in auto_login
                loginCallback=loginCallback, exitCallback=exitCallback)
        File "/usr/lib/python3.7/site-packages/efb_wechat_slave/vendor/itchat/components/login.py", line 56, in login
                status = self.check_login()
        File "/usr/lib/python3.7/site-packages/efb_wechat_slave/vendor/itchat/components/login.py", line 141, in check_login
                r = self.s.get(url, params=params, headers=headers)
        File "/usr/lib/python3.7/site-packages/requests/sessions.py", line 555, in get
                return self.request('GET', url, **kwargs)
        File "/usr/lib/python3.7/site-packages/requests/sessions.py", line 542, in request
                resp = self.send(prep, **send_kwargs)
        File "/usr/lib/python3.7/site-packages/requests/sessions.py", line 655, in send
                r = adapter.send(request, **kwargs)
        File "/usr/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
                timeout=timeout
        File "/usr/lib/python3.7/site-packages/urllib3/connectionpool.py", line 706, in urlopen
                chunked=chunked,
        File "/usr/lib/python3.7/site-packages/urllib3/connectionpool.py", line 440, in _make_request
                httplib_response = conn.getresponse()
        File "/usr/lib/python3.7/http/client.py", line 1369, in getresponse
                        File "/usr/lib/python3.7/http/client.py", line 310, in begin
                        File "/usr/lib/python3.7/http/client.py", line 271, in _read_status
                        File "/usr/lib/python3.7/socket.py", line 589, in readinto
                        File "/usr/lib/python3.7/ssl.py", line 1071, in recv_into
                        File "/usr/lib/python3.7/ssl.py", line 929, in read
                ^C2021-03-27 22:08:06,276 [CRITICAL]: ehforwarderbot.__main__ (__main__.stop_gracefully; __main__.py:78)
    Thread report is now stopped, press Ctrl-C for 3 times to force quit EFB.
^C^C^C2021-03-27 22:08:06,972 [ERROR]: ehforwarderbot.__main__ (__main__.stop_gracefully; __main__.py:84)
    5 consequent exit signals detected.
Force exiting EFB without cleaning up.
```
I also tried to use openwrt's daemon manager, procd, to control the process, the result was the same. 
```
EH Forwarder Bot
Version: 2.1.0
Python version:
3.7

Master channel:
    blueset.telegram

Slave channel:
    blueset.wechat

**Comment by terrytw** [comment]
I have tried it on a Debian x64 10.8 machines as well, seems that when QR code is being displayed, the process won't respond to sigterm.

**Comment by blueset** [comment]
Thank you for the report.

This is caused by a too early graceful stop signal handler registration in the startup process which prevent the normal `KeyboardInterrupt` from taking effect before poll thread is ready.

I’ve made an attempt to fix this issue in ea596f2. It should now solve this problem with the fix. If you are willing to, you may want to try upgrade EH Forwarder Bot framework with the version on GitHub, or otherwise you can wait until this fix is published on PyPI.

**Comment by terrytw** [comment]
I tried in on Debian 10.8, it seems to be fixed! Thanks for the swift response!
